### ISSUE TITLE
HttpDate needs formatting to and from integers

### ISSUE THREAD

**Issue by alexbowers** [hyperium/hyper#1062]
HttpDate currently is a wrapper around Tm, and doesn't support building from integers.

I currently have to do something along the lines of:

```rust
let file_modified_date_duration = file_modified_date.duration_since(UNIX_EPOCH);
let timestamp = time::strptime(&file_modified_date_duration.as_secs().to_string(), "%s");
HttpDate(file_modified_time); // works
```

However, I shouldn't have to format as string, passing it through as a unix timestamp.

```
file_modified_date_duration.as_secs()
```

comes back as a `u64`, which should be acceptable for `HttpDate`

**Comment by seanmonstar** [comment]
So, it is possible currently without going through serializing and parsing a format string, but it is also more work than a single method. How to do so today:

```
// with a Duration
let ts = time::Timespec {
    sec: duration.as_secs() as i64,
    nsec: 0
};
let tm = time::at_utc(ts);
let date = HttpDate(tm);
```

**Comment by seanmonstar** [comment]
Revisiting this, I'm considering removing the (public) usage of the `time` crate to create `HttpDate`. It would still be used internally, for `strftime` and `strptime`, but it could be made to work entirely with `std::time::SystemTime` now.

```rust
headers.set(LastModified(file_modified_date.into()));
```

**Comment by alexbowers** [comment]
that would be great

**Comment by seanmonstar** [comment]
`HttpDate` now has `From<SystemTime>` implemented for it, in v0.11.

**Comment by alexbowers** [comment]
Great, thanks.
